internal/dcache/rpc/server/handler.go

```diff
@@ -801,10 +801,18 @@ func (mv *mvInfo) updateComponentRVs(componentRVs []*models.RVNameAndState, forc
 				}
 
 				if oldState == string(dcache.StateOffline) && newState == string(dcache.StateOutOfSync) {
+					//
 					// Same RV (now online) being reused by fix-mv.
+					// Note: we don't check if the oldstate is inband-offline here, since in case of
+					//       inband-offline state, same RV is not reused by the fix-mv workflow.
+					//
 					continue
 				}
 
+				// For same RV, we cannot have old state as inband-offline.
+				common.Assert(oldState != string(dcache.StateInbandOffline),
+					mv.mvName, oldName, oldState, newState)
+
```

Comment:
let's keep inband-offline state limited to the WriteMV path.. in the handler code let's always use offline.

Reply:
Updated, please review